src/common/com/intellij/plugins/haxe/ide/refactoring/introduce/HaxeIntroduceHandler.java

```diff
@@ -319,15 +324,23 @@ private static boolean notFunctionMethodClass(HaxeComponentType type) {
     if (expression instanceof HaxeCallExpression) {
       final HaxeExpression callee = ((HaxeCallExpression)expression).getExpression();
       text = callee.getText();
+    } else if (expression instanceof HaxeRegularExpression) {
+      text = "REGEX";
+    } else if (expression instanceof HaxeLiteralExpression) {
+      text = "CONST_" + expression.getText().toUpperCase();
```

Comment:
Regular expressions and string literals generated pretty weird names.  This makes them better.

Reply:
I'm OK with using SnakeCase, if that's more common.  In the regex case, it will turn into REGEX1, REGEX2, etc..  For the literal expression the initial selection, your example will turn into CONST_INTERPOLATED_STRING_DATA_DATE_HOURS and will go through the algorithm I described below, resulting in `CONST_INTERPOLATED_STRING_DATA_DATE_HOURS`, `INTERPOLATED_STRING_DATA_DATE_HOURS`, `STRING_DATA_DATE_HOURS`, `DATA_DATE_HOURS`, `DATE_HOURS`, and `HOURS`, with `HOURS` being the default.  `CONST` only gets used when the expression turns out not to have any trailing three-character groupings.

Do you have some other test cases in mind that I should try out?  I'm happy to write the tests, if you can imagine some places where the algorithm will have problems.  (And I'll definitely try the one you've suggested here.

So, now re-reading, you're thinking that any interpolated values should NOT be included in the potential name list?

Reply:
As far as casing goes, ThisWouldBeCamelCase, and THIS_WOULD_BE_UPPER_SNAKE_CASING.  That's what you mean, right?